internal/extender/resource.go

```diff
@@ -97,25 +98,28 @@ func NewExtender(
 	}
 }
 
-// Start is responsible for starting background goroutines for the SparkSchedulerExtender
+// Start is responsible for starting background tasks for the SparkSchedulerExtender
 func (s *SparkSchedulerExtender) Start(ctx context.Context) {
+	_ = wapp.RunWithFatalLogging(ctx, s.doStart)
+}
+
+func (s *SparkSchedulerExtender) doStart(ctx context.Context) error {
 	if s.checkDemandCRDExists(ctx) {
```

Comment:
I think we want to keep the initial call to `s.checkDemandCRDExists(ctx)` sync, as we want to block checking if the crd exists to start scheduling

Reply:
Went back to something similar to what we had previously to explicitly show that we want this check to run once and then launch the rest of the logic in a goroutine.